recipe/patches/0004-Windows-shared-libs.patch

```diff
@@ -0,0 +1,588 @@
+@@ -33,6 +33,7 @@
+ 
+ #include "base/kaldi-types.h"
+ #include "base/kaldi-utils.h"
++#include "base/kaldi-dll-exports.h"
+ /* Important that this file does not depend on any other kaldi headers. */
+ 
+ #ifdef _MSC_VER
+@@ -54,7 +55,7 @@ void SetProgramName(const char *basename);
+ 
+ /// This is set by util/parse-options.{h,cc} if you set --verbose=? option.
+ /// Do not use directly, prefer {Get,Set}VerboseLevel().
+-extern int32 g_kaldi_verbose_level;
++extern int32 KALDI_DLL g_kaldi_verbose_level;
+ 
+ /// Get verbosity level, usually set via command line '--verbose=' switch.
+ inline int32 GetVerboseLevel() { return g_kaldi_verbose_level; }
+diff --git a/src/base/kaldi-types.h b/src/base/kaldi-types.h
+index 68d5578..3a1b8b9 100644
+--- a/src/base/kaldi-types.h
++++ b/src/base/kaldi-types.h
+@@ -39,16 +39,30 @@ typedef float   BaseFloat;
+ // we find in the future lacks stdint.h
+ #include <stdint.h>
+ 
+-typedef int8_t   int8;
+-typedef int16_t  int16;
+-typedef int32_t  int32;
+-typedef int64_t  int64;
+-
+-typedef uint8_t  uint8;
+-typedef uint16_t uint16;
+-typedef uint32_t uint32;
+-typedef uint64_t uint64;
+-typedef float    float32;
+-typedef double   double64;
++namespace fst {
++  typedef int8_t   int8;
++  typedef int16_t  int16;
++  typedef int32_t  int32;
++  typedef int64_t  int64;
++  typedef uint8_t  uint8;
++  typedef uint16_t uint16;
++  typedef uint32_t uint32;
++  typedef uint64_t uint64;
++}  // end namespace fst
++
++
++namespace kaldi {
++  typedef int8_t   int8;
++  typedef int16_t  int16;
++  typedef int32_t  int32;
++  typedef int64_t  int64;
++  typedef uint8_t  uint8;
++  typedef uint16_t uint16;
++  typedef uint32_t uint32;
++  typedef uint64_t uint64;
++  typedef float    float32;
++  typedef double   double64;
++}  // end namespace kaldi
++
+ 
+ #endif  // KALDI_BASE_KALDI_TYPES_H_
+diff --git a/src/bin/phones-to-prons.cc b/src/bin/phones-to-prons.cc
+index 0d7ab12..d6a4422 100644
+--- a/src/bin/phones-to-prons.cc
++++ b/src/bin/phones-to-prons.cc
+@@ -28,9 +28,9 @@
+ 
+ // Create FST that accepts the phone sequence, with any number
+ // of word-start and word-end symbol in between each phone.
+-void CreatePhonesAltFst(const std::vector<int32> &phones,
+-                        int32 word_start_sym,
+-                        int32 word_end_sym,
++void CreatePhonesAltFst(const std::vector<fst::int32> &phones,
++                        fst::int32 word_start_sym,
++                        fst::int32 word_end_sym,
+                         fst::VectorFst<fst::StdArc> *ofst) {
+   using fst::StdArc;
+   typedef fst::StdArc::StateId StateId;
+@@ -172,7 +172,7 @@ int main(int argc, char *argv[]) {
+         if (g_kaldi_verbose_level >= 2) {
+           KALDI_LOG << "phn2word FST is below:";
+           fst::FstPrinter<StdArc> fstprinter(phn2word, NULL, NULL, NULL, false, true, "\t");
+-          fstprinter.Print(&std::cerr, "standard error");
++          fstprinter.Print(std::cerr, "standard error");
```

Comment:
This change looks unmotivated to me, but not critical. The patch is already much smaller, mercifully.

Reply:
It fixes the following that was introduced with a change in OpenFst 1.8:

```
C:\Users\michael\Documents\Dev\mmcauliffe_kaldi\src\bin\phones-to-prons.cc(175): error C2664: 'void fst::FstPrinter<fst::StdArc>::Print(std::ostream &,const std::string &)': cannot convert argument 1 from 'std::ostream *' to 'std::ostream &'
C:\Users\michael\miniconda3\envs\kalpy_build\Library\include\fst/script/print-impl.h(59): note: see declaration of 'fst::FstPrinter<fst::StdArc>::Print'
```